auth/server/base/src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -893,7 +895,19 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                         if (stateRef.get().canVerifyEvidence()) {
                             if (peerCerts != null) {
                                 log.tracef("Authentication ID is null but SSL peer certificates are available. Trying to authenticate peer");
-                                verifyEvidence(new X509PeerCertificateChainEvidence(peerCerts));
+                                State state = stateRef.get();
+                                String mechanismName = state.getMechanismInformation().getMechanismName();
+                                String mechanismType = state.getMechanismInformation().getMechanismType();
+                                // if SASL EXTERNAL mechanism used with skip-certificate-verification property then do not verifyEvidence against the security realm
+                                if ((mechanismType.equals("SASL") && mechanismName.equals("EXTERNAL") && saslSkipCertificateVerification)) {
```

Comment:
This still feels a bit too mechanism specific to be in here - can't saslSkipCertificateVerification be enough on it's own?